bundles/mapping/mapmodule/request/map.layer.handler.js

```diff
@@ -64,6 +67,41 @@ Oskari.clazz.define('map.layer.handler',
                 evt._creator = request._creator;
                 sandbox.notifyAll(evt);
             }
+        },
+        _loadLayerInfo: function (layer, addLayer) {
+            if (typeof layer.getDescribeLayerStatus !== 'function') {
+                // layer type doesn't support this
+                addLayer();
+                return;
+            }
+            const layerId = layer.getId();
+            if (isNaN(layerId)) {
+                // only layers that have numeric ids can have reasonable response for DescribeLayer
+                addLayer();
+                return;
+            }
+            const status = layer.getDescribeLayerStatus();
+            if (status === DESCRIBE_LAYER.LOADED || status === DESCRIBE_LAYER.PENDING) {
+                return;
+            }
+            layer.getDescribeLayerStatus(DESCRIBE_LAYER.PENDING);
+            this.layerService.getDescribeLayer(layerId, info => {
+                if (!info) {
+                    layer.setDescribeLayerStatus(DESCRIBE_LAYER.ERROR);
+                    if (layer.requiresDescripeLayer()) {
+                        // notify user here or in implementations
+                        this.log.error('Attempt to add layer that requires more info. Skipping id: ' + layerId);
+                        return;
+                    }
+                }
+                layer.getDescribeLayerStatus(DESCRIBE_LAYER.LOADED);
+                const sandbox = this.layerService.getSandbox();
+                layer.handleDescribeLayer(info);
+                sandbox.findRegisteredModuleInstance('MainMapModule').handleDescribeLayer(layer, info);
```

Comment:
I think we should move `layer.handleDescribeLayer(info)` to `mapmodule.handleDescribeLayer()` so all the setting of it happens in one place. Or maybe we should set the info to layer here and not pass it to mapmodule with the layer 🤔 

I think this handler class is created in mapmodule so we could also pass a reference to mapmodule in constructor instead of getting it through sandbox, but I'll tinker with it a bit to see what feels best.